File: 11-container-with-most-water/container-with-most-water.ts
function maxArea(height: number[]): number {
    let l = 0
    let r = height.length -1;
    let ret = 0
    while (l<r){
        if (height[r]<height[l]){
            ret = Math.max(ret,(r-l)*height[r])
            r--;
        }
        else{
            ret = Math.max(ret,(r-l)*height[l])
            l++;
        }
    }

    return ret;

};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r = 0,len(height)-1
        l_max,r_max = height[l],height[r]
        ret = 0
        while(l<r):
            if l_max<r_max:
                ret = max(ret,(r-l)*l_max)
                l+=1
                l_max = max(l_max,height[l])
            else:
                ret = max(ret,(r-l)*r_max)
                r-=1
                r_max = max(r_max,height[r])
        return ret